hikari/impl/stateful_cache.py

```diff
@@ -1598,3 +1605,32 @@ def update_voice_state(
         cached_voice_state = self.get_voice_state(voice_state.guild_id, voice_state.user_id)
         self.set_voice_state(voice_state)
         return cached_voice_state, self.get_voice_state(voice_state.guild_id, voice_state.user_id)
+
+    def clear_messages(self) -> cache.CacheView[snowflakes.Snowflake, messages.PartialMessage]:
+        if not self._message_entries:
+            return cache_utility.EmptyCacheView()
+
+        messages = self._message_entries
```

Comment:
might want to do 
```python
cached_messages = self._message_entries.copy
self._message_entries.clear()
```
if you want to preserve the max_messages without tracking it